server/api/crud/alerts.py

```diff
@@ -375,6 +385,20 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
         self._get_alert_state_cached().cache_remove(session, alert.id)
         self._clear_alert_states(alert)
 
+    @staticmethod
+    def _should_reset_alert(old_alert_data, alert_data, force_reset):
+        if force_reset:
+            return True
+
+        # if one of these fields was modified then we should reset the alert
```

Comment:
You are commenting what you are doing and not why.
```suggestion
        # the alert should be reset if a functional parameter was changed, which could affect the decision on when to trigger the alert
```